<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ViewClass</class>
 <widget class="QMainWindow" name="ViewClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1071</width>
    <height>790</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Vcode</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>20</y>
      <width>460</width>
      <height>701</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QGraphicsView" name="leftOriginView">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>50</y>
       <width>256</width>
       <height>192</height>
      </rect>
     </property>
    </widget>
    <widget class="QPushButton" name="recognizeButton">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>270</y>
       <width>90</width>
       <height>30</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 12pt &quot;Agency FB&quot;;</string>
     </property>
     <property name="text">
      <string>开始识别</string>
     </property>
     <property name="shortcut">
      <string>Ctrl+B</string>
     </property>
    </widget>
    <widget class="QTextBrowser" name="resultText">
     <property name="geometry">
      <rect>
       <x>110</x>
       <y>430</y>
       <width>261</width>
       <height>201</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 75 12pt &quot;Courier&quot;;</string>
     </property>
    </widget>
    <widget class="QLabel" name="label">
     <property name="geometry">
      <rect>
       <x>190</x>
       <y>370</y>
       <width>90</width>
       <height>30</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 12pt &quot;Agency FB&quot;;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string>识别结果</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>20</y>
      <width>460</width>
      <height>701</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QGraphicsView" name="removeBgView">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>350</y>
       <width>256</width>
       <height>81</height>
      </rect>
     </property>
    </widget>
    <widget class="QGraphicsView" name="binaryzationView">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>460</y>
       <width>256</width>
       <height>81</height>
      </rect>
     </property>
    </widget>
    <widget class="QGraphicsView" name="grayView">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>240</y>
       <width>256</width>
       <height>81</height>
      </rect>
     </property>
    </widget>
    <widget class="QGraphicsView" name="denoiseView">
     <property name="geometry">
      <rect>
       <x>160</x>
       <y>570</y>
       <width>256</width>
       <height>81</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_4">
     <property name="geometry">
      <rect>
       <x>35</x>
       <y>360</y>
       <width>101</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 14pt &quot;Agency FB&quot;;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string>去背景</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_3">
     <property name="geometry">
      <rect>
       <x>35</x>
       <y>470</y>
       <width>101</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 14pt &quot;Agency FB&quot;;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string>二值化</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_2">
     <property name="geometry">
      <rect>
       <x>35</x>
       <y>580</y>
       <width>101</width>
       <height>51</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 14pt &quot;Agency FB&quot;;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="text">
      <string>去噪点</string>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
    </widget>
    <widget class="QLabel" name="label_5">
     <property name="geometry">
      <rect>
       <x>35</x>
       <y>260</y>
       <width>101</width>
       <height>50</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 14pt &quot;Agency FB&quot;;</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::Box</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="lineWidth">
      <number>1</number>
     </property>
     <property name="text">
      <string>灰度</string>
     </property>
     <property name="textFormat">
      <enum>Qt::AutoText</enum>
     </property>
     <property name="alignment">
      <set>Qt::AlignCenter</set>
     </property>
     <property name="wordWrap">
      <bool>false</bool>
     </property>
    </widget>
    <widget class="QLabel" name="label_6">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>65</y>
       <width>81</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 12pt &quot;Agency FB&quot;；</string>
     </property>
     <property name="text">
      <string>灰度方式</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_7">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>100</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 12pt &quot;Agency FB&quot;；</string>
     </property>
     <property name="text">
      <string>去背景阈值</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_8">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>135</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 12pt &quot;Agency FB&quot;；</string>
     </property>
     <property name="text">
      <string>二值化阈值</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_9">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>170</y>
       <width>170</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 12pt &quot;Agency FB&quot;；</string>
     </property>
     <property name="text">
      <string>去噪点窗口半边长</string>
     </property>
    </widget>
    <widget class="QSlider" name="rBGSlider">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>100</y>
       <width>160</width>
       <height>20</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>150</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="invertedAppearance">
      <bool>false</bool>
     </property>
     <property name="invertedControls">
      <bool>false</bool>
     </property>
     <property name="tickPosition">
      <enum>QSlider::NoTicks</enum>
     </property>
    </widget>
    <widget class="QSlider" name="binarySlider">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>135</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>220</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSlider" name="denoiseSlider">
     <property name="geometry">
      <rect>
       <x>200</x>
       <y>170</y>
       <width>160</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
    <widget class="QSpinBox" name="rBGSpinBox">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>100</y>
       <width>46</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>150</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="binarySpinBox">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>135</y>
       <width>46</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="value">
      <number>220</number>
     </property>
    </widget>
    <widget class="QSpinBox" name="denoiseSpinBox">
     <property name="geometry">
      <rect>
       <x>380</x>
       <y>170</y>
       <width>46</width>
       <height>22</height>
      </rect>
     </property>
     <property name="minimum">
      <number>0</number>
     </property>
     <property name="maximum">
      <number>10</number>
     </property>
    </widget>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>35</y>
       <width>111</width>
       <height>21</height>
      </rect>
     </property>
     <property name="styleSheet">
      <string notr="true">font: 12pt &quot;Agency FB&quot;；</string>
     </property>
     <property name="text">
      <string>验证码类型 </string>
     </property>
    </widget>
    <widget class="QFrame" name="frame">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>25</y>
       <width>281</width>
       <height>31</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QRadioButton" name="slideCodeButton">
      <property name="geometry">
       <rect>
        <x>140</x>
        <y>10</y>
        <width>131</width>
        <height>19</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 12pt &quot;Agency FB&quot;;</string>
      </property>
      <property name="text">
       <string>滑动验证码</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="textCodeButton">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>141</width>
        <height>19</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 12pt &quot;Agency FB&quot;;</string>
      </property>
      <property name="text">
       <string>文字验证码</string>
      </property>
     </widget>
    </widget>
    <widget class="QFrame" name="frame_2">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>57</y>
       <width>341</width>
       <height>41</height>
      </rect>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Raised</enum>
     </property>
     <widget class="QRadioButton" name="aveButton">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>10</y>
        <width>115</width>
        <height>19</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 12pt &quot;Agency FB&quot;;</string>
      </property>
      <property name="text">
       <string>平均值法</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="weightAveButton">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>10</y>
        <width>115</width>
        <height>19</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 12pt &quot;Agency FB&quot;;</string>
      </property>
      <property name="text">
       <string>加权平均</string>
      </property>
     </widget>
     <widget class="QRadioButton" name="maxbutton">
      <property name="geometry">
       <rect>
        <x>110</x>
        <y>10</y>
        <width>115</width>
        <height>19</height>
       </rect>
      </property>
      <property name="styleSheet">
       <string notr="true">font: 12pt &quot;Agency FB&quot;;</string>
      </property>
      <property name="text">
       <string>最大值法</string>
      </property>
     </widget>
    </widget>
    <zorder>frame</zorder>
    <zorder>removeBgView</zorder>
    <zorder>binaryzationView</zorder>
    <zorder>grayView</zorder>
    <zorder>denoiseView</zorder>
    <zorder>label_4</zorder>
    <zorder>label_3</zorder>
    <zorder>label_2</zorder>
    <zorder>label_5</zorder>
    <zorder>label_6</zorder>
    <zorder>label_7</zorder>
    <zorder>label_8</zorder>
    <zorder>label_9</zorder>
    <zorder>rBGSlider</zorder>
    <zorder>binarySlider</zorder>
    <zorder>denoiseSlider</zorder>
    <zorder>rBGSpinBox</zorder>
    <zorder>binarySpinBox</zorder>
    <zorder>denoiseSpinBox</zorder>
    <zorder>label_10</zorder>
    <zorder>frame_2</zorder>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>2</width>
      <height>2</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout"/>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1071</width>
     <height>26</height>
    </rect>
   </property>
   <widget class="QMenu" name="file">
    <property name="title">
     <string>文件</string>
    </property>
    <property name="toolTipsVisible">
     <bool>false</bool>
    </property>
    <addaction name="importPicAction"/>
    <addaction name="saveAction"/>
    <addaction name="separator"/>
    <addaction name="exitAction"/>
   </widget>
   <widget class="QMenu" name="help">
    <property name="title">
     <string>帮助</string>
    </property>
    <addaction name="donateAction"/>
    <addaction name="guideAction"/>
    <addaction name="separator"/>
    <addaction name="aboutAction"/>
   </widget>
   <addaction name="file"/>
   <addaction name="help"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="importPicAction">
   <property name="text">
    <string>导入图片</string>
   </property>
  </action>
  <action name="exitAction">
   <property name="text">
    <string>退出</string>
   </property>
  </action>
  <action name="donateAction">
   <property name="checkable">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>捐助</string>
   </property>
  </action>
  <action name="aboutAction">
   <property name="text">
    <string>关于</string>
   </property>
  </action>
  <action name="guideAction">
   <property name="text">
    <string>指南</string>
   </property>
  </action>
  <action name="saveAction">
   <property name="text">
    <string>保存文件</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="Vcode.qrc"/>
 </resources>
 <connections/>
</ui>
